src/ffi.md

```diff
@@ -737,11 +737,11 @@ void foo(struct Foo *arg);
 void bar(struct Bar *arg);
 ```
 
-To do this in Rust, let’s create our own opaque types with `enum`:
+To do this in Rust, let’s create our own opaque types:
 
 ```rust
-pub enum Foo {}
-pub enum Bar {}
+#[repr(C)] pub struct Foo { private: [u8; 0] }
+#[repr(C)] pub struct Bar { private: [u8; 0] }
```

Comment:
I think this should be `_priv:` to avoid unused field warnings, and to match stdlib style.

I also don't really understand the point of repr(C) here? Silencing the lint?

Reply:
Yes, rustc warns without it. Same with `[u8; 0]` over `()`.